### PR TITLE
Add readme to examples.

### PR DESCRIPTION
**Pull request by thorstenhater**
Add a readme for the examples, aimed at beginners, who might want to learn Arbor's C++ interface by looking at the examples.

### CODE REVIEW

**Comment by noraabiakar** [comment]
I think it would be good to also mention how to build the c++ examples in the docs, maybe before [this](https://arbor.readthedocs.io/en/latest/install/build_install.html#quick-start-examples) section? 
Also the dryrun readme is wrong, can we also fix that at the same time?

**Comment by thorstenhater** [comment]
Good points, done.

**Comment by brenthuisman** [comment]
Does this PR fix #868?

**Comment by thorstenhater** [comment]
By accident, yes ;)

File: example/readme.md [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+All examples can be built by setting up the standard
+CMake build process, e.g.
+
+    # go to arbor's source directory   
+    cd arbor
+    # set up a build dir
+    mkdir build
+    cd build
+    # configure (example options for an MPI-enabled build)
```
Comment (noraabiakar) [inline example/readme.md:15]: Can we mention that it's also vectorization-enabled here? And below, instructions how to run, with a note saying to check the readme of each example?

**Review by noraabiakar — COMMENTED** [review]
Looks good! thanks for also fixing the dryrun readme. I have just one small suggestion.

File: example/readme.md (hunk 1 above)
Comment (thorstenhater) [reply example/readme.md:15]: Certainly I can add SIMD here. But the individual READMEs are referenced above this example.

Comment (thorstenhater) [reply example/readme.md]: That said, I made it a bit more clear ... hopefully.

**Review by noraabiakar — APPROVED** [review]

**Pull request merged**